Update developer OAuth integration documentation (#22682)
DataDog/documentation: content/en/developers/integrations/oauth_for_integrations.md

## Overview

> OAuth enables Datadog customers to easily and securely authorize third-party access to their Datadog organization. This authorization allows integrations to push data into Datadog or pull data out from Datadog without the need for customers to input API or app keys anywhere. For example, a user can consent to provide an on-call notification tool read access to their Datadog organization's monitors.

Reviewer: ```suggestion
OAuth enables Datadog customers to securely authorize third-party access to their Datadog organization. This authorization allows integrations to push data into Datadog or pull data out from Datadog without the need for customers to input API or app keys anywhere. For example, a user can consent to provide an on-call notification tool with read access to their Datadog organization's monitors.
```

For more information on Datadog's OAuth implementation, see the [Datadog OAuth2 documentation][1].

## When to use OAuth in an integration

OAuth support is required for all partner-built SaaS integrations that directly submit data to, or query data from, Datadog's public [API endpoints][12]. OAuth does not apply to software deployed on-premises, or to Datadog Agent checks. 

## Build an integration with OAuth

> When building an integration with OAuth, you should only select the scopes to which your application needs access. Once a customer consents to authorize your integration, all listed scopes will become available to your application via a token.

Reviewer: ```suggestion
When building an integration with OAuth, you should only select the scopes to which your application needs access. After a customer consents to authorize your integration, all listed scopes become available to your application through a token.
```
